cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -111,12 +115,18 @@ public void setUp() throws Exception {
       doClear(configuredNamespace, false);
     }
     registeredNamespaces.put(configuredNamespace, deleteUponTeardown);
+    applicationManagers.clear();
     LOG.info("Completed setUp.");
   }
 
   @After
   public void tearDown() throws Exception {
     LOG.info("Beginning tearDown.");
+
+    // Stop any running programs created in @Test
+    for (ApplicationManager manager: applicationManagers) {
```

Comment:
The reason that the next test cases can fail is that a thread is still fetching `program.status.event.topic` messages from TMS that pertain to a program that was deleted in the previous test.

When the thread tries to write to the store that the program has been killed, an error is thrown from the store (rightfully so) because there was no record of that program ever starting (since it was deleted from the last test).